plugins/catalog-backend/src/ingestion/processors/UrlReaderProcessor.ts

```diff
@@ -78,4 +85,32 @@ export class UrlReaderProcessor implements CatalogProcessor {
 
     return true;
   }
+
+  private async doRead(
+    location: string,
+  ): Promise<{ data: Buffer; url: string }[]> {
+    // Does it contain globs? I.e. does it contain asterisks or question marks
+    // (no curly braces for now)
+    const { filepath } = parseGitUrl(location);
+    if (filepath?.match(/[*?]/)) {
+      const response = await this.options.reader.search(location);
+      const output = response.files.map(async file => {
+        const url = new URL(location);
+
+        // pathname starts as /backstage/backstage/blob/master/<path>
+        const base = url.pathname.split('/').slice(1, 5).join('/');
+        url.pathname = `${base}/${file.path}`;
```

Comment:
This is only true for GitHub right? or at least not necessarily true for GitLab and Azure. We could use the new resolve utility from integrations with an absolute path right?

Could fix in a later PR since we'll ofc only get here for GitHub, but a TODO could be good